app.go

```diff
@@ -911,5 +942,5 @@ func (a *Application) ReadOnlyState() State {
 		a.lastBlockHeader,
 		nil, // TODO: last block hash!
 		a.GetValidatorSet,
-	)
+	).WithEVMState(snapshotEVMState)
```

Comment:
This should probably be something like:
```golang
).WithEVMState(a.EvmStore.GetSnapshot().StateDB())
```
Also `a.EvmStore.GetVMRootKey()` as currently implemented is not go-routine-safe.